src/snapshot.rs

```diff
@@ -1,6 +1,37 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
 use crate::libdeno::deno_buf;
 
+#[cfg(target_arch = "aarch64")]
+use crate::flags::v8_set_flags;
+#[cfg(target_arch = "aarch64")]
+use crate::libdeno::deno_generate_snapshot;
+#[cfg(target_arch = "aarch64")]
+use std::ffi::CString;
+
+#[cfg(target_arch = "aarch64")]
+pub fn deno_snapshot() -> deno_buf {
+  let js_file = "gen/bundle/main.js";
+  #[cfg(not(feature = "check-only"))]
+  let js_source =
+    include_bytes!(concat!(env!("GN_OUT_DIR"), "/gen/bundle/main.js"));
+  // The snapshot blob is not available when the Rust Language Server runs
+  // 'cargo check'.
+  #[cfg(feature = "check-only")]
+  let js_source = vec![];
+
+  let cjs_file = CString::new(js_file).unwrap();
+  let cjs_source =
+    CString::new(std::str::from_utf8(js_source).unwrap()).unwrap();
+
+  v8_set_flags(vec![
+    "gen/snapshot_deno.bin".to_string(),
+    js_file.to_string(),
+  ]);
+
+  unsafe { deno_generate_snapshot(cjs_file.as_ptr(), cjs_source.as_ptr()) }
```

Comment:
This is generating the snapshot at runtime, which seems not so ideal. I think just executing the code instead would work..?

Reply:
I want to save the snapshot to disk on first run, so it can be loaded from disk on subsequent runs.

Reply:
Hm... but then why not just include it in the executable?
We should try to get snapshot_creator to build as a target_arch...

Reply:
I guess we need something like this to accomplish that:
https://github.com/denoland/deno_third_party/blob/7e697531c16ee6713fa3d986447bbd4dc04a80bb/v8/BUILD.gn#L3492

Reply:
> Hm... but then why not just include it in the executable?
 We should try to get snapshot_creator to build as a target_arch...

How would one go about executing the snapshot generator on the compiler host if the snapshot generator is compiled for a different architecture.